app/src/main/java/com/byoutline/kickmaterial/KickMaterialApp.kt

```diff
@@ -25,28 +39,35 @@ class KickMaterialApp : Application() {
         AndroidStubServer.start(this, NetworkType.UMTS)
         RetrofitHelper.setMsgDisplayer { msg -> this.showToast(msg, true) }
 
-        resetComponents()
+        initDagger()
     }
 
     @VisibleForTesting
-    @Synchronized
-    fun setComponents(mainComponent: GlobalComponent) {
-        component = mainComponent
-        Settings.set(debug = BuildConfig.DEBUG, containerViewId = R.id.container)
+    open fun initDagger() {
+        setComponents(createGlobalComponent())
     }
 
-    private fun resetComponents() {
-        val mainComponent = createGlobalComponent()
-        setComponents(mainComponent)
+    @VisibleForTesting
+    open fun setComponents(appComponent: AppComponent) {
+        component = appComponent
+        component.inject(this)
+        AppInjector.init(this)
+        SecretSauceSettings.set(debug = BuildConfig.DEBUG,
+                containerViewId = R.id.container,
+                bindingViewModelId = BR.viewModel,
```

Comment:
Databinding name of ViewModel to be set after inflating. Needed only for `inflateAndSetViewModel` extension function. If not set it will have to be passed on each `inflateAndSetViewModel` making it probably longer then old way of calling Databinding directly 
```
val binding = ...Binding.inflate...
binding.viewModel = viewModel
```